### DISCUSSION TITLE
Link Google account with email and password sign-in?

### DISCUSSION THREAD

**Discussion by alvinsga** [pocketbase/pocketbase#1709] [Q&A] [answered]
On my app there are two options to login, one using Google OAuth and another using email and password. I noticed that once a user logs in with Google, they are not able to create an account with email and password sign-in. I get the error "The email is invalid or already in use". Is there a way to link the two auth providers together to give users the flexibility to choose the way they login?

**Comment by ganigeorgiev** [comment, marked as answer]
On OAuth2 sign-up we create a new auth record (aka. user) with the email returned by the provider (if any) and a random generated password. On first sign up, if there was a user with the same email returned by the provider, the accounts will be "linked" automatically.

You can't create a new auth record with the same email. If you want to allow your users to login with email after the OAuth2 sign-up, you can send them a password reset/set email (eg. `pb.collection("users").requestPasswordReset(email)`).

**Reply by ganigeorgiev** [reply]
I forgot to mention that if the user account was initially created with email/password, then you also have the option to "link" the OAuth2 user if the `authWithOAuth2()` request was made with authenticated `pb.authStore` and if the OAuth2 user is not already linked to another auth record, eg:

```js
pb.collection("users").authWithPassword("test@example.com", "1234567890");

// authenticate with OAuth2 and link the OAuth2 user to the previously authenticated auth record
pb.collection("users").authWithOAuth2(...)
```

**Reply by shubhendumadhukar** [reply]
This might be a silly question, but I was going through the example provided for OAuth2 login in the [documentation](https://pocketbase.io/docs/authentication/#web-oauth2-integration). While implementing something similar for my app, I am getting the value of `params.get('code')` as null. If I understood it correctly the resulting URI from `provider.authUrl + redirect_url` should have a query param `code` with some value, but I just don't see any such query parameter. Would you please help me point to what I might be missing?

For reference, this is what I get if I console log `provider.authUrl + redirect_url`
```
https://accounts.google.com/o/oauth2/auth?client_id=<<client_id>>&code_challenge=<<code_challenge>>
&code_challenge_method=<<code_challenge_method>>&response_type=code
&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile+https%3A%2F%2Fwww.googleapis.com
%2Fauth%2Fuserinfo.email&state=<<state>>&redirect_uri=<<redirect_uri>>
```

**Reply by ganigeorgiev** [reply]
@shubhendumadhukar 

`provider.authUrl + redirect_url` constructs a url pointing to the providers sign-in/sign-up page.

The `code` query parameter should be available **after** the provider redirects the user to your specified redirect url.
